File: src/instagram.rs

pub struct Instagram {
    base_url: String,
}

pub struct Post {}

impl<'str> Instagram {
    pub fn new() -> Instagram {
        Instagram {
            base_url: String::from("https://graph.instagram.com/me/media"),
        }
    }

    pub async fn get_posts<'a>(&self) -> std::vec::Vec<Post> {
        let access_token = self.get_token();

        let complete = format!(
            "{}?fields={}&access_token{}",
            self.base_url.clone(),
            "id,media_type,media_url,thumbnail_url,username,timestamp",
            access_token
        );

        vec![Post {}]
    }

    fn get_token(&self) -> String {
        String::from("IGQVJXQkQ5UGIzeTN2RlQ3NHB1SjZAfZAGt4NHZATV0ZAjQVVaXzUtdnByaWFEYnRPbHJ2QjhiSWxfVXRJQ3JMU0JoLUZAWbUJEZA2JiU2djdTRyYk9XXzZATby1UemRuVnJGMFNGb2ZApaGFKUTl5WGNKZAl9UdwZDZD")
    }
}

File: Instagram.py
import requests
from logging import debug, info, error

INSTAGRAM_TOKEN = ""

with open("./.instagram_token", encoding="utf-8") as f:
    INSTAGRAM_TOKEN = f.readline()


def get_user_media():
    global INSTAGRAM_TOKEN

    fields = "id" # caption(hashtag)
    url = "https://graph.instagram.com/me/media?fields={}&access_token={}".format(fields, INSTAGRAM_TOKEN)
    result = requests.get(url=url)
    print(url)
    print("get user media result: ", result.json())

    return result.json()


def get_representative_media_id(media_id: str):
    global INSTAGRAM_TOKEN

    fields = "id" # timestamp
    url = "https://graph.instagram.com/{}?fields={}&access_token={}".format(media_id, fields, INSTAGRAM_TOKEN)
    result = requests.get(url=url)
    debug("get_representative_media_id result: ", result.json())

    return result.json()["id"]


def get_media_children(child_id: str):
    global INSTAGRAM_TOKEN

    fields = "id,media_url" # timestamp
    url = "https://graph.instagram.com/{}/children?fields={}&access_token={}".format(child_id, fields, INSTAGRAM_TOKEN)
    result = requests.get(url=url)
    debug("child result: ", result.json())

    return result.json()


# Use this method ONLY!!!
def get_photo_url():
    media = get_user_media()
    urls = []
    for e in media["data"]:
        e_id = get_representative_media_id(e["id"])
        datas = get_media_children(e_id)["data"]
        for data in datas:
            print("data: ", data)
            urls.append(data["media_url"])